app/controllers/workshops_controller.rb

```diff
@@ -93,7 +93,7 @@ def workshop
 	end
 
 	def workshop_params
-		params.require(:workshop).permit(:title, :description, :user_id, :slug, :image, :topic_id, :duration, :price, :format, :expected_outcomes)
+		params.require(:workshop).permit(:title, :description, :user_id, :slug, :image, :topic, :format, :price, :duration, :topic_id, :format_id, :price_id, :duration_id, :expected_outcomes)
```

Comment:
what attributes that you don't use, you have to remove it from the permit params, in your case, you MUST remove :price, :format and :duration because use will be using id. I took care of this.